plugin/db/mongodb/mongodb.go

```diff
@@ -0,0 +1,139 @@
+	connectionURI := getMongoDBConnectionURI(connCfg)
+	client, err := mongo.NewClient(options.Client().ApplyURI(connectionURI))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create MongoDB client")
+	}
+	err = client.Connect(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to MongoDB")
+	}
+	driver.client = client
+	return driver, nil
+}
+
+// Close closes the MongoDB driver.
+func (driver *Driver) Close(ctx context.Context) error {
+	if err := driver.client.Disconnect(ctx); err != nil {
+		return errors.Wrap(err, "failed to disconnect MongoDB")
+	}
+	return nil
+}
+
+// Ping pings the database.
+func (driver *Driver) Ping(ctx context.Context) error {
+	if err := driver.client.Ping(ctx, nil); err != nil {
+		return errors.Wrap(err, "failed to ping MongoDB")
+	}
+	return nil
+}
+
+// GetDBConnection returns a database connection.
+func (*Driver) GetDBConnection(_ context.Context, _ string) (*sql.DB, error) {
+	panic("not implemented")
+}
+
+// Execute executes a statement.
+func (*Driver) Execute(_ context.Context, _ string, _ bool) (int64, error) {
+	panic("not implemented")
+}
+
+// Query queries a statement.
+func (*Driver) Query(_ context.Context, _ string, _ *db.QueryContext) ([]interface{}, error) {
+	panic("not implemented")
+}
+
+// SyncInstance syncs the instance meta.
+func (*Driver) SyncInstance(_ context.Context) (*db.InstanceMeta, error) {
+	panic("not implemented")
+}
+
+// SyncDBSchema syncs the database schema.
+func (*Driver) SyncDBSchema(_ context.Context, _ string) (*db.Schema, error) {
+	panic("not implemented")
+}
+
+// NeedsSetupMigration returns whether the driver needs to setup migration.
+func (*Driver) NeedsSetupMigration(_ context.Context) (bool, error) {
+	panic("not implemented")
+}
+
+// SetupMigrationIfNeeded sets up migration if needed.
+func (*Driver) SetupMigrationIfNeeded(_ context.Context) error {
+	panic("not implemented")
+}
+
+// ExecuteMigration executes a migration.
+func (*Driver) ExecuteMigration(_ context.Context, _ *db.MigrationInfo, _ string) (int64, string, error) {
+	panic("not implemented")
+}
+
+// FindMigrationHistoryList finds the migration history list.
+func (*Driver) FindMigrationHistoryList(_ context.Context, _ *db.MigrationHistoryFind) ([]*db.MigrationHistory, error) {
+	panic("not implemented")
+}
+
+// Dump dumps the database.
+func (*Driver) Dump(_ context.Context, _ string, _ io.Writer, _ bool) (string, error) {
+	panic("not implemented")
+}
+
+// Restore restores the backup read from src.
+func (*Driver) Restore(_ context.Context, _ io.Reader) error {
+	panic("not implemented")
+}
+
+// getMongoDBConnectionURI returns the MongoDB connection URI.
+// https://www.mongodb.com/docs/manual/reference/connection-string/
+func getMongoDBConnectionURI(connConfig db.ConnectionConfig) string {
+	connectionURL := "mongodb://"
+	if connConfig.SRV {
+		connectionURL = "mongodb+srv://"
+	}
+	if connConfig.Username != "" {
```

Comment:
Need to use `percent encoding` for username and password according to the docs?

Reply:
Good catch